apps/namadillo/src/App/Common/TableWithPaginator.tsx

```diff
@@ -0,0 +1,102 @@
+import { StyledTable, TableHeader, TableRow } from "@namada/components";
+import { FormattedPaginator } from "App/Common/FormattedPaginator";
+import clsx from "clsx";
+import { useCallback, useEffect, useMemo, useRef, useState } from "react";
+
+type TableWithPaginatorProps<T> = {
+  id: string;
+  headers: (TableHeader | React.ReactNode)[];
+  itemList: T[];
+  renderRow: (item: T, index: number) => TableRow;
+  resultsPerPage?: number;
+  initialPage?: number;
+  children?: React.ReactNode;
+} & Pick<React.ComponentProps<typeof StyledTable>, "tableProps" | "headProps">;
+
+export const TableWithPaginator = <T,>({
```

Comment:
Just a question: Is this now generic enough to live in @namada/components? I don't think it's important to move it soon, if at all, just a thought!

Reply:
I was wondering about this too. I'm not 100% sure it's not too interface specific yet and it does depend on the react-paginate dependency, so I left it here for now, but I definitely think this could be considered for a move to components in the future.